### IMDB Project, Part 2
#### Laura Dimaria
8/1/22

Note: I am only able to extract year 2001, so I completed the assignments with the year I had and will come back when the issue is fixed.

In [1]:
import pandas as pd
import numpy as np

In [2]:
year_2001 = pd.read_csv('/Users/lauradimaria/Desktop/GitHub/CD--Project-3-IMDB-/Data/final_tmdb_data_2001.csv.gz')
year_2001 = year_2001.drop(labels=0, axis=0)
year_2001.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1284 entries, 1 to 1284
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1284 non-null   object 
 1   adult                  1284 non-null   float64
 2   backdrop_path          649 non-null    object 
 3   belongs_to_collection  86 non-null     object 
 4   budget                 1284 non-null   float64
 5   genres                 1284 non-null   object 
 6   homepage               108 non-null    object 
 7   id                     1284 non-null   float64
 8   original_language      1284 non-null   object 
 9   original_title         1284 non-null   object 
 10  overview               1256 non-null   object 
 11  popularity             1284 non-null   float64
 12  poster_path            1127 non-null   object 
 13  production_companies   1284 non-null   object 
 14  production_countries   1284 non-null   object 
 15  rele

### How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

In [3]:
year_2001 = year_2001[(year_2001['revenue'] > 0.0) | (year_2001['budget'] > 0.0)]
year_2001.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 328 entries, 1 to 1265
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                328 non-null    object 
 1   adult                  328 non-null    float64
 2   backdrop_path          265 non-null    object 
 3   belongs_to_collection  50 non-null     object 
 4   budget                 328 non-null    float64
 5   genres                 328 non-null    object 
 6   homepage               52 non-null     object 
 7   id                     328 non-null    float64
 8   original_language      328 non-null    object 
 9   original_title         328 non-null    object 
 10  overview               326 non-null    object 
 11  popularity             328 non-null    float64
 12  poster_path            313 non-null    object 
 13  production_companies   328 non-null    object 
 14  production_countries   328 non-null    object 
 15  relea

328 movies had valid financial information in either budget or revenue, down from 1284.

### Drop remainder movies with 0's for budget AND revenue

In [4]:
year_2001.drop(year_2001.loc[year_2001['revenue'] == 0.0].index, inplace=True)
year_2001.drop(year_2001.loc[year_2001['budget'] == 0.0].index, inplace=True)
year_2001.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 183 entries, 1 to 1242
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                183 non-null    object 
 1   adult                  183 non-null    float64
 2   backdrop_path          176 non-null    object 
 3   belongs_to_collection  41 non-null     object 
 4   budget                 183 non-null    float64
 5   genres                 183 non-null    object 
 6   homepage               37 non-null     object 
 7   id                     183 non-null    float64
 8   original_language      183 non-null    object 
 9   original_title         183 non-null    object 
 10  overview               183 non-null    object 
 11  popularity             183 non-null    float64
 12  poster_path            183 non-null    object 
 13  production_companies   183 non-null    object 
 14  production_countries   183 non-null    object 
 15  relea

183 movies have some financial data for both revenue and budget

### How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [5]:
year_2001['certification'].value_counts()

R        75
PG-13    62
PG       14
G         5
NR        2
Name: certification, dtype: int64

### What is the average revenue per certification category?

In [6]:
#code stolen from stack overflow
#link: https://stackoverflow.com/questions/21137150/format-suppress-scientific-notation-from-pandas-aggregation-results
pd.set_option('display.float_format', lambda x: f'{x:,.3f}')

year_2001.groupby('certification')['revenue'].mean()

certification
G       177,814,925.400
NR       26,808,346.000
PG      180,331,769.429
PG-13   109,245,343.968
R        54,561,884.827
Name: revenue, dtype: float64

### What is the average budget per certification category?

In [7]:
pd.set_option('display.float_format', lambda x: f'{x:,.3f}')
year_2001.groupby('certification')['budget'].mean()

certification
G       42,400,000.000
NR       7,200,000.000
PG      53,239,286.071
PG-13   45,514,473.984
R       30,033,120.667
Name: budget, dtype: float64